### PR TITLE
Feature/update decommissioning

### PR DESCRIPTION
**Pull request by ClaraBuettner**
This branch allows reducing the capacity of all components using decommissioning scenarios.

### CODE REVIEW

File: etrago/tools/io.py [hunk 1]
```diff
@@ -788,45 +790,99 @@ def decommissioning(network, session, args, **kwargs):
...
-    query = session.query(ormclass).filter(
-                        ormclass.scn_name == 'decommissioning_' + 
-                        args['scn_decommissioning'])
+            
+        if version== None:   
+            ormclass = getattr(import_module('egoio.db_tables.model_draft'),
+                           'EgoGridPfHvExtension' + comp)
+        else:
+            ormclass = getattr(import_module('egoio.db_tables.grid'),
+                           'EgoPfHvExtensionLine' + comp)
```
Comment (ulfmueller) [inline etrago/tools/io.py]: in line 800 and 803 it seems wrong that it is '...HvExtension' and '...HvExtensionLine' ?!

Comment (ClaraBuettner) [reply etrago/tools/io.py]: Yes, that was wrong, I changed it in my latest commit

**Comment by ClaraBuettner** [comment]
With these commits, all decommissioning scenarios on oedb work well. I think the branch can be merged.

File: etrago/appl.py [hunk 2]
```diff
@@ -449,10 +449,15 @@ def etrago(args):
...
-            network,
-            session,
-            args)
+        for i in range(len(args['scn_decommissioning'])):
+            network = decommissioning(
+                    network,
+                    session,
+                    version=args['gridversion'],
+                    scn_decommissioning=args['scn_decommissioning'][i],
+                    start_snapshot=args['start_snapshot'],
```
Comment (lukasoldi) [inline etrago/appl.py]: Hi, here you define start_snapshot and end_snapshot for decommissioning(), but these arguments are not required by the function which leads to an error.

Comment (ClaraBuettner) [reply etrago/appl.py]: Hi, thank you for your help. I fixed this in the last commit.

File: etrago/tools/io.py [hunk 3]
```diff
@@ -788,45 +790,94 @@ def decommissioning(network, session, args, **kwargs):
...
+    
+    if comp =='Line':
+         comp_df = network.lines[network.lines.index.isin(df_decommisionning.index)]
+         new_cap = comp_df.s_nom-df_decommisionning.s_nom
+         network.lines.s_nom[network.lines.index.isin(df_decommisionning.index)]=\
+         new_cap
+         
+         for idx, row in network.lines.iterrows():
+             if (row['s_nom_min'] !=0) & (
+                        row['scn_name'] =='extension_' + scn_decommissioning):
```
[the remark this answers is not in the document]
Comment (ClaraBuettner) [reply etrago/tools/io.py]: I know this error, but with commit 6246a89 I solved the problem (missing brackets). So I don't have this error anymore.

**Comment by ClaraBuettner** [comment]
The error ERROR:pypsa.io:Error, new components for Carrier are not unique is caused by the extension scenarios (Carriers are added for each scenario, with the same ids and the same name). But everything works well with the error message, all carriers are there and all generators are correctly assigned, I tested this. I tried to solve the problem several times but couldn't find a solution.

**Comment by lukasoldi** [comment]
Have you tested the decommissioning scenarios with a k-means clustering? While the extension seems to work fine, I receive an error when decommissioning. Have you seen something similar?

**Comment by ClaraBuettner** [comment]
Yes, I tested it with k-means (and [this pypsa branch](https://github.com/openego/PyPSA/tree/feature/pypsa_13_fork)) without any errors.

**Comment by ClaraBuettner** [comment]
There still seem to be some issues here so changes and testing are needed before merging. 
Removing e.g. generation capacities by decommissioning-scenarios is still a nice feature in my opinion. Maybe we leave the PR open and I will implement this later?

**Comment by ulfmueller** [comment]
If we do not include this, the decommissioning function does not work, right? I think it would be nice if all args functions work for the release at least in some 'basic' way. It sounds like that you were happy with the branch but only Lukas had problems...

**Comment by ClaraBuettner** [comment]
the decommissioning function works, but only for ac-lines. With these changes, all components can be decommissioned or get reduced capacities with decommissioning scenarios

**Comment by ulfmueller** [comment]
ok. let's leave it out for now then.

**Pull request closed without merging**